tests/integration_test/websocket_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration_test
```

Comment:
I think we need a test for the service control report on websockets. I'm curious what it will look like.

Reply:
Report is only called when the websocket connection is closed, I think. All data passing byte will not go through our filters,  so we don't know how many bytes passing byte,  we only count the first request/response

Reply:
Thanks Wayne for the explaination.